src/ModelContextProtocol/Client/StreamableHttpClientSessionTransport.cs

```diff
@@ -203,16 +203,40 @@ private async Task ReceiveUnsolicitedMessagesAsync()
         return null;
     }
 
-    private void LogJsonException(JsonException ex, string data)
+    /// <summary>
+    /// Sends the initial initialization request and returns the HTTP response so the status code can be checked.
+    /// </summary>
+    /// <param name="message">The initialize message to send, which must be a JsonRpcRequest with the method "initialize".</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The HTTP response message for the initialization request.</returns>
+    internal async Task<HttpResponseMessage> SendInitialRequestAsync(
```

Comment:
SendInitialRequestAsync and SendMessageAsync should share code. The implementation of each should be a couple lines at most because they call a shared method. The only difference is that SendInitialRequestAsync holds on to the HttpResponseMessage and returns it rather than dispose it.